tests/framework/utils.py

```diff
@@ -206,3 +229,106 @@ def get_cpu_vendor():
     if 'AuthenticAMD' in brand_str:
         return CpuVendor.AMD
     return CpuVendor.INTEL
+
+
+def is_range(content):
+    """Return true if `content` is a range.
+
+    E.g ranges: 0-10.
+    """
+    match = re.search("([0-9][1-9]*)-([0-9][1-9]*)", content)
+    # Group is a singular value.
+    return match is not None
+
+
+def list_range(content):
+    """Return a range of integers based on the `content`.
+
+    The `content` respects the LIST FORMAT defined in the
+    cpuset documentation.
+    See: https://man7.org/linux/man-pages/man7/cpuset.7.html.
+    """
+    content = content.strip()
+    ends = content.split("-")
+    if len(ends) != 2:
+        return None
+
+    return list(range(int(ends[0]), int(ends[1]) + 1))
+
+
+def parse_list_format(content):
+    """Parse list formats for cpuset and mems.
+
+    See LIST FORMAT here: https://man7.org/linux/man-pages/man7/cpuset.7.html.
+    """
+    content = content.strip()
+    if len(content) == 0:
+        return []
+
+    groups = content.split(",")
+    arr = set()
+
+    def func(acc, cpu):
+        if is_range(cpu):
+            acc.update(list_range(cpu))
+        else:
+            acc.add(int(cpu))
+        return acc
+
+    return list(functools.reduce(func, groups, arr))
+
+
+def cpus_arr():
```

Comment:
The name can be hard to understand. `cpsu_array` or `cpus` sounds better to me.

Reply:
Since this method is only used by the `CpuMap` class, could it be moved inside the class?

Reply:
Moved the method to `CpuMap` and renamed it as well to `cpus`.